multimodal/src/autogluon/multimodal/predictor.py

```diff
@@ -713,23 +713,29 @@ def fit(
             fit_called=fit_called,
         )
 
-        self._problem_type = self._infer_problem_type(train_data=train_data, column_types=column_types)
-
         if tuning_data is None:
             train_data, tuning_data = self._split_train_tuning(
                 data=train_data, holdout_frac=holdout_frac, random_state=seed
             )
 
-        column_types = self._infer_column_types(
-            train_data=train_data, tuning_data=tuning_data, column_types=column_types
+        if self._label_column:
+            self._problem_type = infer_problem_type(
+                y_train_data=train_data[self._label_column],
+                provided_problem_type=self._problem_type,
+            )
+
+        column_types = infer_column_types(
+            data=train_data,
+            valid_data=tuning_data,
+            label_columns=self._label_column,
+            provided_column_types=column_types,
+            problem_type=self._problem_type,  # used to update the corresponding column type
         )
 
-        # FIXME: separate infer problem_type with output_shape, should be logically distinct
-        _, output_shape = infer_problem_type_output_shape(
+        output_shape = infer_output_shape(
```

Comment:
Move `infer_output_shape` to be directly after getting the problem type, since they are logically sequential, and generally unrelated to `infer_column_types`

Reply:
Make sense. Will switch the location in next PRs.